changelog/dpath.dd

```diff
@@ -0,0 +1,41 @@
+DUB settings & packages directory placement overhauled
+
+You can now configure where DUB places its downloaded packages and where the user configuration is stored through environment variables or through the dub configuration. You need to use an environment variable or the system-wide dub configuration to specify where the user configuration is stored.
+
+By default DUB stores the packages on
+- Windows: `%APPDATA%/dub/{packages/,settings.json}`
+- Posix: `$HOME/.dub/{packages/,settings.json}`
+
+now if the `DUB_HOME` environment variable is set it instead stores the packages (and other config) in
+- `$DUB_HOME/{packages/,settings.json}`
+
+alternatively if `DUB_HOME` isn't set, but `DPATH` is set, the following path is used:
+- `$DPATH/dub/{packages/,settings.json}`
+
+The `DPATH` environment variable is intended to be used by all D tooling related things doing user-space installation of things. It can be used to avoid cluttering the home folder.
+
+Additionally to environment variables it is possible to configure the package placement path + settings.json path through DUB's settings.json file. To configure where the user-editable settings.json is placed you need to adjust the system-wide dub configuration.
+
+In the settings.json you can set the following fields:
+
+```json
+{
+    "dubHome": "/path/to/dub", // sets both package store and config location
+
+    // OR
+
+    "userSettings": "/path/to/dub/settings/folder",
+    "localRepository": "/path/to/dub/repository/folder"
+}
+```
+
+Additionally, these config paths will have environment variables using the `$VARIABLE` syntax resolved.
+
+The following list describes which path is going to be picked, from top to bottom, stopping whenever one is found:
+
+- `$DUB_HOME` environment variable
+- `$DPATH` environment variable
+- system-wide settings.json: `"userSettings"` property
+- system-wide settings.json: `"dubHome"` property (only for userSettings)
+- through all settings.json: `"localRepository"` property
+- through all settings.json: `"dubHome"` property (only for localRepository)
```

Comment:
this overrides the built-in dub location, so it would never even attempt to read anything from the built-in path, while for customCachePaths it's just an addition of paths.